## Arturo Ramos Viedas
# A01636133

# Undertanding the iris data set

The notebook aims to undertand the content of the iris data set.


## Acknowledgments

- Data from https://www.coursera.org/ from the course "Understanding and Visualizing Data with Python" by University of Michigan


# Importing and inspecting the data

In [ ]:
# Define where you are running the code: colab or local
RunInColab          = True    # (False: no  | True: yes)

# If running in colab:
if RunInColab:
    # Mount your google drive in google colab
    from google.colab import drive
    drive.mount('/content/drive')

    # Find location
    #!pwd
    #!ls
    #!ls "/content/drive/My Drive/Colab Notebooks/MachineLearningWithPython/"

    # Define path del proyecto
    Ruta            = "/content/drive/My Drive/A01636133/"

else:
    # Define path del proyecto
    Ruta            = ""

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# Import the packages that we will be using
import matplotlib.pyplot as plt
import pandas as pd

# Dataset url
url = Ruta + "datasets/iris/iris.csv"
colnames = ['Sepal length', 'Sepal width', 'Petal length', 'Petal width', 'Class']

# Load the dataset
dataset = pd.read_csv(url, names=colnames, header=None)


In [ ]:
# Print the dataset
dataset

Sepal length  Sepal width  Petal length  Petal width           Class
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

In [ ]:
# Print the number of rows
Nrows = dataset.shape[0]
Nrows

150

In [ ]:
# Print the number of columns
Ncols = dataset.shape[1]
Ncols

5

# Data types

In [ ]:
dataset.dtypes

Sepal length    float64
Sepal width     float64
Petal length    float64
Petal width     float64
Class            object
dtype: object

# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and understand the dataset

2. How many observations (rows) are in total?
150

3. How many variables (columns) are in total? What do they represent?
5

4. How many observations are for each type of flower?
50

5. What is the type of data for each variable?
<br>Sepal length: Float
<br>Sepal width: Float
<br>Petal length: Float
<br>Petal width: Float
<br>Class: Object

6. What are the units of each variable?
<br>Sepal length: cm
<br>Sepal width: cm
<br>Petal length: cm
<br>Petal width: cm
<br>Class: N/A
